# Enable kFreeBSD and Hurd specific configure script
---
# configure |    2 +-
# tmux.h    |   16 ++++++++++++++++
# 2 files changed, 17 insertions(+), 1 deletion(-)
#
--- a/tmux.h
+++ b/tmux.h
@@ -396,6 +396,18 @@
 	char		argv[COMMAND_LENGTH];
 };
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 64
+#endif
+
 struct msg_identify_data {
 	char		cwd[MAXPATHLEN];
 
@@ -1114,6 +1126,10 @@
 };
 RB_HEAD(status_out_tree, status_out);
 
+#ifndef IOV_MAX
+#define IOV_MAX	1024
+#endif
+
 /* Client connection. */
 struct client {
 	struct imsgbuf	 ibuf;
--- a/osdep-freebsd.c
+++ b/osdep-freebsd.c
@@ -29,7 +29,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#if 0
 #include <libutil.h>
+#endif
 
 struct kinfo_proc	*cmp_procs(struct kinfo_proc *, struct kinfo_proc *);
 char			*osdep_get_name(int, char *);
@@ -135,6 +137,7 @@
 char *
 osdep_get_cwd(pid_t pid)
 {
+#if 0
 	static char		 wd[PATH_MAX];
 	struct kinfo_file	*info = NULL;
 	int			 nrecords, i;
@@ -151,6 +154,7 @@
 	}
 
 	free(info);
+#endif
 	return (NULL);
 }
 
